### PR TITLE
woodwork 0.31.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-4688](https://anaconda.atlassian.net/browse/PKG-4688) 
- [Upstream repository](https://github.com/alteryx/woodwork/tree/v0.31.0)
- Requirements: https://github.com/alteryx/woodwork/blob/v0.31.0/pyproject.toml

### Explanation of changes:

- Reset the build number to `0.`
- Update pinnings in `run` and `test/requires`

[PKG-4688]: https://anaconda.atlassian.net/browse/PKG-4688?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,11 +31,11 @@ requirements:
     - wheel
   run:
     - importlib_resources >=5.10.0
-    - numpy >=1.22.0,<1.25.0
-    - pandas >=1.4.3
+    - numpy >=1.25.0
```
Comment (ryanskeith) [inline recipe/meta.yaml]:
```suggestion
    - numpy >=1.25.0, <2.0.0
```
per line https://github.com/alteryx/woodwork/blob/e2f0adadd9371858f0838f264f3fd755040c8b31/pyproject.toml#L38

**Review by ryanskeith — COMMENTED** [review]
Please change the upper bound on numpy after that it looks good to me.

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: That's not correct. They removed the upper bound in `>=0.29.0`, see https://github.com/alteryx/woodwork/commit/a32ac361ce3f251bc5801a4bb3c43e3eff3ffff2

Comment (cbouss) [reply recipe/meta.yaml]: They added it back after the last release though. If you look at https://github.com/alteryx/woodwork/compare/v0.31.0...main

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Thank you. Fair enough. I fixed it.

**Review by cbouss — APPROVED** [review]

**Review by lorepirri — APPROVED** [review]

**Pull request merged**